server/build-plugins.js

```diff
@@ -33,7 +42,74 @@ const getPackageName = name => {
   }
 };
 
-const prepareModules = async (plugins = [], local = true) => {
+async function installRemotePackages(installPackages) {
+  // check if connected to internet
+  // if not, skip npm install
+  const EXTERNAL_URI = process.env.EXTERNAL_URI || 'npmjs.com';
+  await require('dns').lookup(EXTERNAL_URI, async err => {
+    if (err && err.code === 'ENOTFOUND')
+      logger.error(`Can't reach npm servers. Skipping npm install`);
+    else {
+      logger.info('Installing plugin packages...');
+      execSync(
+        `npm install --no-save ${installPackages.join(' ')} --production`,
```

Comment:
This seems like it could be an attack vector, wonder if this is a place where somebody could somehow execute an arbitrary bash script